building/Windows.md

```diff
@@ -0,0 +1,133 @@
+```
+
+After all of that, we need to run Cygwin installation because it will download all the needed dependencies without
+needing to do this manually.
+
+The next step is to prepare the genext2fs library and clone a repository. We need to make manually genext2fs library
+because Cygwin doesn't support it, and it's mandatory in case of creating images for the ia32 system.
+
+To make it simply paste this into the Cygwin terminal (Same as before we want to use commands line by line):
+
+``` bash
+git clone --recursive https://github.com/bestouff/genext2fs
+
+cd genext2fs
+
+./autogen.sh
+./configure
+make
+make install
+make check
+
+cd
+rm -rf genext2fs apt-cyg
+
+git clone --recursive https://github.com/phoenix-rtos/phoenix-rtos-project/
+```
+
+Now we can simply follow instructions about building toolchains remembering to use commands to build inside
+phoenix-rtos-project folder.
+(Quick tip: It is not mandatory but turning off Real-Time Protection in Windows Defender speeds up this process)
+
+``` bash
+(cd phoenix-rtos-build/toolchain/ && ./build-toolchain.sh i386-pc-phoenix ~/toolchains/i386-pc-phoenix)
+(cd phoenix-rtos-build/toolchain/ && ./build-toolchain.sh arm-phoenix ~/toolchains/arm-phoenix)
+(cd phoenix-rtos-build/toolchain/ && ./build-toolchain.sh riscv64-phoenix ~/toolchains/riscv64-phoenix)
+(cd phoenix-rtos-build/toolchain/ && ./build-toolchain.sh sparc-phoenix ~/toolchains/sparc-phoenix)
+```
+
+Adding toolchains binaries to the PATH variable also can be tricky. On Windows, we move over folders with names
+containing white spaces, after using:
+
+``` bash
+echo "export PATH=$PATH:$HOME/toolchains/i386-pc-phoenix/i386-pc-phoenix/bin/:$HOME/toolchains/arm-phoenix/arm-phoenix/bin/:$HOME/toolchains/riscv64-phoenix/riscv64-phoenix/bin/:$HOME/toolchains/sparc-phoenix/sparc-phoenix/bin/" >> $HOME/.bashrc
+```
+
+We need to fix paths e.g. `C:\x\y z\d` we must change to `C:\x\'y z'\d` simple change will fix all unwanted messages at
+the start of the Cygwin terminal and will allow us to use the toolchain without setting the path every time.
+
+To test is recommended to use this build command:
+
+``` bash
+TARGET=ia32-generic-qemu CONSOLE=serial phoenix-rtos-build/build.sh clean core fs image project test
+```
+
+And to run QEMU with this image we can use
+
+``` bash
+./scripts/ia32-generic-qemu-test.sh
+```
+
+## VSL (Windows Subsystem for Linux)
+
+WSL is based on a Linux virtualization environment on a Windows environment.
+To properly install WSL is needed to allow virtualization in BIOS settings as default they are disabled.
+
+WSL we can easily download via
+
+```powershell
+winget install wsl
+```
+
+In PowerShell (Remember always run as administrator).

```

Comment:
I would add information about running PowerShell as an administrator above the code block.